src/rust/en.tsumino/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+#![no_std]
+
+use aidoku::{
+	error::Result,
+	helpers::uri::encode_uri,
+	prelude::*,
+	std::net::HttpMethod,
+	std::net::Request,
+	std::{String, Vec},
+	Chapter, DeepLink, Filter, FilterType, Listing, Manga, MangaPageResult, Page,
+};
+extern crate alloc;
+use alloc::{string::ToString, vec};
+mod helper;
+mod parser;
+
+#[get_manga_list]
+fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+	let mut query: String = String::new();
+	let mut sort: String = String::from("Newest");
+	let mut tags: String = String::new();
+	let mut i = 0;
+	for filter in filters {
+		match filter.kind {
+			FilterType::Title => {
+				query.push_str("&Text=");
+				query.push_str(&encode_uri(filter.value.as_string()?.read()));
+			}
+			FilterType::Genre => {
+				let tpe = 1;
+				tags.push_str(&format!(
+					"&Tags[{}][Type]={}",
+					i.to_string(),
+					tpe.to_string()
+				));
```

Comment:
what is `tpe` for? I think you  can simplify to
```suggestion
				tags.push_str(&format!("&Tags[{i}][Type]=1"));
```